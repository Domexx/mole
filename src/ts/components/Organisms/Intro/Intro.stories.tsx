import { Story } from '@storybook/react/types-6-0';
import React, { ComponentProps } from 'react';
import IntroComponent from './Intro';

const Component = {
  title: 'Components/Organisms/Intro',
  component: IntroComponent,
  argTypes: {},
};

const Template: Story<ComponentProps<typeof IntroComponent>> = (args) => {
  return <IntroComponent {...args} />;
};

export const Intro = Template.bind({});
export default Component;
