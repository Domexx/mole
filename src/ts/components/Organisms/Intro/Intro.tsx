import React, { useContext } from 'react';
import MenuItem from '@mui/material/MenuItem';
import Select from '@mui/material/Select';
import { GameContext, GameContextData } from '../../../states/GameStateProvider';
import Button from '../../Atoms/Button';
import { GameStage } from '../../../states/GameState';

const Intro: React.FC = () => {
  const gameData: GameContextData = useContext(GameContext);

  return (
    <div data-testid="Intro" className="flex flex-col items-center space-y-5 min-w-[10rem]">
      <div className="flex flex-col sm:flex-row space-x-2 items-center">
        <div className="text-center sm:text-right">Select the number of circles to get started:</div>
        <Select
          id="count-select"
          value={gameData.state.CircleCount}
          label="Number of circles"
          onChange={(event) => {
            return gameData.setCircleCount(event.target.value as number);
          }}
          variant="standard"
        >
          <MenuItem value={3}>Three</MenuItem>
          <MenuItem value={5}>Five</MenuItem>
          <MenuItem value={10}>Ten</MenuItem>
        </Select>
      </div>
      <Button
        onClick={() => {
          return gameData.setStage(GameStage.Playing);
        }}
      >
        Start
      </Button>
    </div>
  );
};

export default Intro;
