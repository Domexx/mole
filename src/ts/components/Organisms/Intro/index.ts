import loadable from '@loadable/component';

import './Intro.css';

export default loadable(() => {
  return import('./Intro');
});
