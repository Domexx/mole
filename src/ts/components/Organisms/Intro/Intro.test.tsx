import React from 'react';
import { toJson } from '@test';
import renderer from 'react-test-renderer';
import Intro from './Intro';

test('Matches Snapshot', () => {
  const component = renderer.create(<Intro />);
  const tree = toJson(component);
  expect(tree).toMatchSnapshot();
});
