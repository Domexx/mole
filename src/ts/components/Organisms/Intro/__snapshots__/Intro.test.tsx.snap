// Vitest Snapshot v1

exports[`Matches Snapshot 1`] = `
<div
  className="flex flex-col items-center space-y-5 min-w-[10rem]"
  data-testid="Intro"
>
  <div
    className="flex flex-col sm:flex-row space-x-2 items-center"
  >
    <div
      className="text-center sm:text-right"
    >
      Select the number of circles to get started:
    </div>
    <div
      className="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary  css-149989x-MuiInputBase-root-MuiInput-root-MuiSelect-root"
      onClick={[Function]}
    >
      <div
        aria-expanded="false"
        aria-haspopup="listbox"
        aria-labelledby="count-select"
        className="MuiSelect-select MuiSelect-standard MuiInput-input MuiInputBase-input css-197ijnv-MuiSelect-select-MuiInputBase-input-MuiInput-input"
        id="count-select"
        onBlur={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onMouseDown={[Function]}
        role="button"
        tabIndex={0}
      >
        Three
      </div>
      <input
        aria-hidden={true}
        className="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
        onAnimationStart={[Function]}
        onChange={[Function]}
        tabIndex={-1}
        value={3}
      />
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconStandard css-hd63pv-MuiSvgIcon-root-MuiSelect-icon"
        data-testid="ArrowDropDownIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M7 10l5 5 5-5z"
        />
      </svg>
    </div>
  </div>
</div>
`;
